More Style Changes

AfroThundr3007730 opened the pull request:
This patchset makes the following changes:

Commit 1:
  * Fixed nav-header color (grey, not transparent)
  * Fixed article cards and backgrounds on category pages
  * Consolidated link colors and restored hover color
  * Fixes stying on options page to match the site
  * Fixed help page styling and contrast issues
  * Made search boxes and forms dark as well
  * Fixed up linter errors and warnings

Commit 2:
  * Purges the legacy CSS code from 2014

Commit 3:
  * Added more color variables in lieu of hardcoded values
  * Updated login/signup page styles and buttons for consistency
  * Updated navigation header/footer styles after legacy CSS removal
  * Style consistency tweaks and selector consolidation
  * Apply styles for the previously-overlooked secondary pages
    (e.g.: premium, sponsor, publisher, developer, blog)
  * Fix styling for queue and reader view (this fully fixes #3)

Changes tested on Debian with Firefox and Chrome, both logged in, and in private/incognito mode.

AfroThundr3007730 committed:
More Style Changes:

  - Fixed nav-header color (grey, not transparent)
  - Fixed article cards and backgrounds on category pages
  - Consolidated link colors and restored hover color
  - Fixes stying on options page to match the site
  - Fixed help page styling and contrast issues
  - Made search boxes and forms dark as well
  - Fixed up linter errors and warnings

the-j0k3r commented:
> This should also fix #3 (tested on Debian with Firefox and Chrome)

It should or does? =) if it actually does, the ideal is to add fixes #3 to the commit message (contributing.md) so the fixes are tracked in the commit view easier.

ANd thanks again for your contribution, the only thing I wanted to do and didnt was remove all the original CSS beofre you made your first contribution, because all that CSS is more than dead.

> Fixed up linter errors and warnings

The dev scripts take care of that, or they should.

AfroThundr3007730 commented:
@the-j0k3r Noted. Should I go ahead and drop the old CSS and add back the missing styles (if any)?

the-j0k3r commented on pocket-dark.user.css:
```diff
@@ -259,28 +260,126 @@
+    }
+  }
+  .legacy-sidebar h3, .legacy-sidebar h5, .legacy-sidebar h5>a {
+    color: #478f8f;
```
Ide just add this to variables in :root =)

AfroThundr3007730 replied:
Will do

the-j0k3r replied:
We need to review the root var shades of the colors, they were just ripped off the marketing stylesheet from pocket.

the-j0k3r commented on pocket-dark.user.css:
```diff
@@ -237,11 +238,11 @@
-    background: #111;
-    border: 1px solid #000 !important;
+    background: var(--black);
+    border: 1px solid black !important;
```
I think using :root vars is better here also

the-j0k3r commented on pocket-dark.user.css:
```diff
@@ -237,11 +238,11 @@
   }
   .queue_list .item, .grid .queue_list .item {
-    border-color: #000;
+    border-color: black;
```
I think using :root vars is better here also

AfroThundr3007730 replied:
The normal keyword seemed to make sense in this case. The `--black` variable is already  `#111`. I can rename it to `--almost-black` or make `--true-black: #000;` instead.

the-j0k3r replied:
The darkest should be #111 and lightest #eee really so we can add other shades to account for that and use them as such.

I presume the input fields need to be always darker.

The point of adding root vars (to me) is so we dont have a mix and match, but rather consistency. ;)

the-j0k3r commented on pocket-dark.user.css:
```diff
@@ -185,7 +186,7 @@
     color: var(--white);
   }
-  .coreform-container p > a {
+  .coreform-container p>a {
```
```suggestion
  .coreform-container p > a {
```

AfroThundr3007730 replied:
@the-j0k3r I'm assuming this should also apply to L325 and L367?

the-j0k3r replied:
dont bother here too much, when this is merged,I'll run perfectionist `npm run clean` and these should all be picked up and corrected. though Im wondering about the spaces after colons why they werent fixed by it.

AfroThundr3007730 replied:
Yeah the `npm run clean` missed these when I ran it earlier.

the-j0k3r commented:
> @the-j0k3r Noted. Should I go ahead and drop the old CSS and add back the missing styles (if any)?

Yes, drop the old CSS I think since 2014 is safe to say that stufff had no effect anymore.

>and add back the missing styles (if any)

If theres any actual valid CSS still on there, I think the only valid was `body`. But its not feasible to do a selector by selector test.
What do you think?

AfroThundr3007730 commented:
> If theres any actual valid CSS still on there, I think the only valid was `body`. But its not feasible to do a selector by selector test.
> What do you think?

I'll just remove all the old CSS and add back any missing rules, or add new ones as needed.

the-j0k3r commented:
the helpage looks semi broken too https://help.getpocket.com/

I wasnt able to login with google and Im not going to make some account with Firefox to check what is broken when logged in account is used.
Mind, I dont really use this style, but I dont mind adding a little here and there once in a while.

Once youre done here and this is merged, I'll give it a once over before making a new release so we can officially necromance this style into life.

the-j0k3r commented:
@Mottie I think if @AfroThundr3007730 so wants, to add him as a contributor to this theme.

AfroThundr3007730 committed:
Purging legacy CSS

AfroThundr3007730 committed:
Further style updates

  - Added more color variables in lieu of hardcoded values
  - Updated login/signup page styles and buttons for consistency
  - Updated navigation header/footer styles after legacy CSS removal
  - Style consistency tweaks and selector consolidation
  - Apply styles for the previously-overlooked secondary pages
    (e.g.: premium, sponsor, publisher, developer, blog)
  - Fix styling for queue and reader view (this fully fixes #3)

AfroThundr3007730 commented:
@the-j0k3r Once more, if you please. :)

The latest commit makes quite a few changes to the existing styles and adds a bunch more for the rest of the missing unstyled pages/elements. The consolidated diff can get a bit confusing, so checking the individual diffs might be helpful.

I've left commented section headings in for maintenance purposes, although I can remove them if it's really necessary. I also tried to avoid moving rules around to keep the diff clean(ish). Perhaps another commit to group them properly should follow. This probably rates a major version bump when it lands.

And as my linter informs me, we now have 4 instances of `!important` in the stylesheet. All necessary to override native styles, unfortunately. :(

the-j0k3r commented:
LGTM, =)

This is good to go in now, I think anything missing can be done after.

Ill merge this now and do a release when I get a few moments later on.

Thanks again, looking better =)

lukewlms commented:
@AfroThundr3007730 Excellent - thanks a million for the update!
